jwst/guider_cds/tests/test_guider_cds.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+import pytest
+
+from jwst import datamodels
+from jwst.guider_cds.guider_cds import get_dataset_info, guider_cds
+
+
+@pytest.fixture(scope='module')
+def make_guider_image():
+    """Generate science image"""
+
+    image = datamodels.GuiderRawModel()
+
+    image.meta.instrument.name = 'FGS'
+    image.meta.exposure.frame_time = 234.3423235
+    image.meta.exposure.ngroups = 4
+    image.meta.exposure.group_time = 465.643643
+
+    image.data = np.random.rand(4,10,10,10)
+
+    return image
+
+
+def test_get_dataset_info(make_guider_image):
+    """Make sure information assigned to datamodel is retrieved correctly."""
+
+    model = make_guider_image
+    model.meta.exposure.type = 'FGS_FINEGUIDE'
+
+    imshape, n_int, grp_time, exp_type = get_dataset_info(model)
+
+    assert (imshape == (10,10) and
+            n_int == model.data.shape[0] and
+            grp_time == model.meta.exposure.group_time and
+            exp_type == model.meta.exposure.type)
+
+
+def test_guider_cds_fineguide_mode(make_guider_image):
+    """Test the fine guiding mode
+    """
+
+    model = make_guider_image
+    model.meta.exposure.type = 'FGS_FINEGUIDE'
+
+    truth = np.zeros(model.data.shape)
+
+    result = guider_cds(model)
+
+    n_int = model.data.shape[0]
+    imshape = (model.data.shape[2], model.data.shape[3])
+
+    slope_int_cube = np.zeros((n_int,) + imshape, dtype=np.float32)
+
+    for num_int in range(0, n_int):
+        data_sect = model.data[num_int, :, :, :]
+        first_4 = data_sect[:4, :, :].mean(axis=0)
+        last_4 = data_sect[-4:, :, :].mean(axis=0)
+        slope_int_cube[num_int, :, :] = last_4 - first_4
+
+        truth = slope_int_cube /  model.meta.exposure.group_time
+
+    assert np.array_equal(result.data, truth)
```

Comment:
It might not be safe to check for equality, since these values are float.  You could use `np.allclose` instead.

Reply:
Will make the change, thanks Phil!